src/libs/actions/PersonalDetails.js

```diff
@@ -23,6 +24,26 @@ Onyx.connect({
     callback: val => personalDetails = val,
 });
 
+Onyx.connect({
+    key: ONYXKEYS.MY_PERSONAL_DETAILS,
+    callback: (val) => {
+        if (!val) {
```

Comment:
Besides moving the block i also added this bit so that when we log out if this is null it just returns early since there is no reason to proceed with val is empty.

Reply:
There is one more thing, this should not run when the user is logged out.